File: backend/src/main/java/org/example/backend/domain/fish/dto/FishResponseDto.java
package org.example.backend.domain.fish.dto;

import org.example.backend.domain.fish.entity.Fish;

public record FishResponseDto(
    Long fishId,
    String fishSpecies,
    String fishName
) {

  public FishResponseDto(Fish fish) {
    this(
        fish.getId(),
        fish.getSpecies(),
        fish.getName()
    );
  }
}

File: backend/src/main/java/org/example/backend/domain/fish/dto/FishResponseDto.kt
package org.example.backend.domain.fish.dto

import org.example.backend.domain.fish.entity.Fish

data class FishResponseDto(
    val fishId: Long,
    val fishSpecies: String,
    val fishName: String
) {
    constructor(fish: Fish) : this(
        fish.id,
        fish.species,
        fish.name
    )
}
